the-collab-lab/tcl-22-smart-shopping-list
# Update meta tags

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description



This PR takes care of all the meta information in the app:
* icons
* theme colour
* title
* description
* open graph tags

I did some research into React Helmet and decided not to use it for this task. According to [this issue](https://github.com/nfl/react-helmet/issues/213), React Helmet can only inject basic meta-tags on the client-side. For open graph/schema/twitter tags the app would need to be rendered on the server-side.

The main purpose of this PR is to make the app look good when we share it on LinkedIn and Twitter, so basic meta-tags aren't enough. That's why I updated the tags manually instead (let me know if there is a better solution).

## Related Issue


Closes #49 

## Acceptance Criteria


* New favicon and page title are showing up in the browser
* Theme colour is green (as seen in mobile browser)
* New title and meta description are showing up when you inspect the `<head>` element and when you share the link on social media
* Custom preview is showing up when you share the link on social media (so far it is showing up in Facebook and messaging apps; Twitter and LinkedIn are supposed to be able to fetch images from the production site, once this PR is merged).

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |

## Updates

### Before


![image](https://user-images.githubusercontent.com/68360696/120055269-ea66b800-c002-11eb-8aa1-f9580f65bf1a.png)

### After


![image](https://user-images.githubusercontent.com/68360696/120054980-47616e80-c001-11eb-9959-2f3e07113ee7.png)
![image](https://user-images.githubusercontent.com/68360696/120055377-ae802280-c003-11eb-9e46-c01d20c8d741.png)
![Untitled](https://user-images.githubusercontent.com/68360696/120055588-20a53700-c005-11eb-9111-24530b37bcc9.png)
![photo_2021-05-28_22-36-35](https://user-images.githubusercontent.com/68360696/120055598-3286da00-c005-11eb-9d61-510ab49b0829.jpg)


## Testing Steps / QA Criteria


Use the [preview link](https://tcl-22-shopping-list--pr65-dt-meta-tags-7c04n13h.web.app) to see how it behaves:

* in the browser (title, favicon, meta description)
* in a mobile browser (theme colour)
* when shared on social media (but maybe don't actually share it just yet)

You can also see how [this tool](https://www.opengraph.xyz/) parses the tags.
